server/src/test/java/org/elasticsearch/cluster/ClusterStateTests.java

```diff
@@ -1242,6 +1242,6 @@ public static int expectedChunkCount(ToXContent.Params params, ClusterState clus
             }
         }
 
-        return chunkCount;
```

Comment:
Should we return long instead, avoid casting altogether, and fix the callers to make long-to-long comparison all the way instead?

Reply:
@DaveCTurner made the original change here. I honestly don't know, but I would prefer to keep impact low especially since these are tests.

Reply:
@ldematte I switched from `(int)` to `Math.toIntExact(long);` so that the test throws if things overflow.

If callers see failures here, then we can adjust to return a `long`.

Reply:
Yeah representing chunk counts as `long` everywhere would be fine IMO. I mean it's not going to work very well if these tests actually make more than 2<<31 chunks, but nor do we need to bother the reader with the potential for the down-casting to fail.